include/envoy/server/options.h

```diff
@@ -130,6 +130,11 @@ class Options {
    * @return uint64_t the maximum name length of a stat.
    */
   virtual uint64_t maxStatNameLength() PURE;
+
+  /**
+   * @return uint64_t the maximum name length of a cluster/route/listener.
+   */
+  virtual uint64_t maxUserSuppliedNameLength() PURE;
```

Comment:
I think that it would be better to have this value be derived from the maxStatNameLen, or possibly have that one be derived from this one.  Being able to specify both seems confusing.

Reply:
Yup exactly. No more options are needed. Please rework as @ggreenway suggests.

Reply:
Yeah good point. Does it suffice to just derive max cluster name length as max_stat_name_len - reserved_stat_suffix_len (which happens to be 67 at the moment) ? That would make things considerably simple.

Reply:
Yup that's exactly what I would do.